### PR TITLE
Use Poltergeist as the JS driver, fixes #15

### LINKED ISSUE #15: Use Poltergeist instead of Selenium as a JS driver
This should remove the local dependency on the firefox binary.

### PR DESCRIPTION
**Pull request by awead**
This replaces Selenium with Poltergeist as the JS driver, fixing #15. Travis will be able to execute these tests. However, they are currently failing locally. Once we have them passing, we can remove the `pending` and Travis will run them. If, however, they do not work on Travis but pass locally, we can use the `unless $in_travis` option to do that.

### CODE REVIEW

File: spec/features/unassigned_accession_spec.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 require 'spec_helper'
 
-describe "Creating a new unassigned accession", :js=>true do
```
Comment (cam156) [inline spec/features/unassigned_accession_spec.rb]: I am not sure you want to remove the js: true.  It is a  REAL thing that causes the spec to act differently.  Check out the third code block down (https://github.com/jnicklas/capybara/blob/master/README.md#using-capybara-with-rspec).  it describes using js:true.

File: spec/features/upload_new_file_spec.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 require 'spec_helper'
 require 'features/common'
 
-describe "Creating a new accession", :js=>true do
```
Comment (cam156) [inline spec/features/upload_new_file_spec.rb]: Do not remove js:true (see comment above)

File: spec/features/create_new_accession_spec.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 require 'spec_helper'
 
-describe "Creating a new accession", :js=>true do
```
Comment (cam156) [inline spec/features/create_new_accession_spec.rb]: Do not remove js: true.  It is a real directive to spec capybara, which we still need.  Look here for documentation: https://github.com/jnicklas/capybara/blob/master/README.md#using-capybara-with-rspec

File: spec/features/link_to_bad_accession_spec.rb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 require 'spec_helper'
 
 
-describe "link to a bad accession", :js=>true do
```
Comment (cam156) [inline spec/features/link_to_bad_accession_spec.rb]: Do not remove js:true.  See comment above

**Comment by mjgiarlo** [comment]
Do you want all those `js: true`s removed?  We don't need those to tell the test suite when to spin up the javascript-specific bits?

Otherwise I'm :+1:

File: spec/features/unassigned_accession_spec.rb (hunk 1 above)
Comment (awead) [inline spec/features/unassigned_accession_spec.rb]: @cam156 this is only if you want to specify a different JS driver and we're using poltergeist throughout.

Comment (awead) [inline spec/features/unassigned_accession_spec.rb]: @cam156 also, I get the exact same behavior in the test with or without the js option, so I'm not sure what it's actually doing.

**Comment by awead** [comment]
@mjgiarlo that's what I'm trying to figure out. See my comment to @cam156 above. I find no discernible difference in behavior with or without the `true` option. Scholarsphere doesn't use it either, save for one test, and I know the other tests are relying on JS behaviors to work.

**Comment by awead** [comment]
@mjgiarlo @cam156 ok, it looks like we don't need `js: true` because we're specifying a default driver here:

https://github.com/psu-stewardship/archivesphere/pull/19/files#diff-a88a62fdf8d6308f9cb96289e5068d11R10

 See https://github.com/teampoltergeist/poltergeist/issues/334

**Comment by mjgiarlo** [comment]
Hrmm.  Should we use that as the default driver throughout?  I believe we used to do this in Sufia and then moved to using racktest where possible (because it's faster) and poltergeist where necessary.

**Comment by awead** [comment]
@mjgiarlo I'm basing all of this off of what's currently being dong in Sufia and SS.

**Comment by awead** [comment]
@mjgiarlo In SS, the default poltergeist driver is applied to all feature tests, while the racktest drive would be used elsewhere. Since we're not using Capybara anywhere else outside of the feature tests here, it wouldn't apply.

**Pull request merged**